sourcegraph/sourcegraph
# Code insights: add dashboard delete modal

Closes https://github.com/sourcegraph/sourcegraph/issues/22232

### Background
This PR adds delete dashboard modal UI and delete dashboard logic.

By click on this action item, you must see a modal UI with delete UI. By click on the delete forever button you will be redirected to the dashboard owner built-in dashboard (user, org or global dashboard)

**Context delete action item at the dashboard page** 
<img width="598" alt="Screenshot 2021-07-13 at 21 57 19" src="https://user-images.githubusercontent.com/18492575/125509451-b462587f-3425-40ec-9119-db39522d64bb.png">



**Dashboard delete modal UI**
<img width="786" alt="Screenshot 2021-07-13 at 21 57 31" src="https://user-images.githubusercontent.com/18492575/125509465-dbdf31b3-f759-4ff9-9906-25be7251f785.png">

### How to test
1. Go to /insights/dashboards
2. Create a new dashboard by create dashboard button
3. Open dashboard 3-dots context menu and pick delete actions
4. You must see delete modal 
5. Delete Dashboard forever 
6. See built-in dashboard and see that you don't have a deleted dashboard in dashboard select
